301-availability-set-elastic-storage-ecs/azuredeploy.json

```diff
@@ -0,0 +1,726 @@
+    },
+    "dnsPrefix": {
+      "type": "string",
+      "defaultValue": "[concat('ecs', uniquestring(resourceGroup().id))]",
+      "metadata": {
+        "description": "dns name prefix. this is the external dns prefix to reach your ECS via the loadbalancer"
+      }
+    },
+    "osImagePublisher": {
+      "type": "string",
+      "defaultValue": "OpenLogic",
+      "allowedValues": [
+        "Centos",
+        "OpenLogic"
+      ],
+      "metadata": {
+        "description": "The Image Publisher, use Openlogic for default Marketplace"
+      }
+    },
+    "osImageOffer": {
+      "type": "string",
+      "defaultValue": "CentOS",
+      "allowedValues": [
+        "Centos-7",
+        "CentOS"
+      ],
+      "metadata": {
+        "description": "The Image Offer, use centos for default Marketplace"
+      }
+    },
+    "osImageSKU": {
+      "type": "string",
+      "defaultValue": "7.4",
+      "allowedValues": [
+        "Centos-7.4",
+        "7.4"
+      ],
+      "metadata": {
+        "description": "The Linux version for the VM. Use 7.4 for default Marketplace"
+      }
+    }
+  },
+  "variables": {
+    "availabilitySetName": "[toLower(concat('ECSaSet-', resourceGroup().name))]",
+    "storageAccountType": "Standard_LRS",
+    "vmSize": "[parameters('vmSize')]",
+    "dnsPrefix": "[parameters('dnsPrefix')]",
+    "osImageVersion": "latest",
+    "addressPrefix": "10.0.0.0/16",
+    "virtualNetworkName": "[tolower(concat('vNet-',resourceGroup().name))]",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworkName'))]",
+    "NICPrefix": "vnic-",
+    "subnetPrefix": "10.0.0.0/24",
+    "subnetName": "vmstaticsubnet",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
+    "storageName": "[concat('sa', uniquestring(resourceGroup().id))]",
+    "loadBalancerProbeName": "443Probe",
+    "loadBalancerBackendName": "LoadBalancerBackend",
+    "publicLBName": "[tolower(concat('external-lb-', resourceGroup().name))]",
+    "publiclbID": "[resourceId('Microsoft.Network/loadBalancers',variables('publicLBName'))]",
+    "lbFE": "[tolower(concat('external-lb-fe-',resourceGroup().name))]",
+    "publiclbFEConfigID": "[concat(variables('publiclbID'),'/frontendIPConfigurations/',variables('lbFE'))]",
+    "publicIPAddressName": "[tolower(concat('public-ip',resourceGroup().name))]",
+    "nsgName": "[tolower(concat('vmnsg',resourceGroup().name))]",
+    "nsgID": "[resourceId('Microsoft.Network/networkSecurityGroups',variables('nsgName'))]",
+    "vmContainerName": "vhds",
+    "diagnosticsStorageAccountName": "[concat('diag', uniquestring(resourceGroup().id))]",
+    "diagnosticsStorageAccountType": "Standard_LRS",
+    "lbPorts": [
+      443,
+      2049,
+      9020,
+      9021,
+      9022,
+      9023,
+      9024,
+      9025
+    ],
+    "gitUrl": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/301-availability-set-elastic-storage-ecs"
```

Comment:
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc) for how to handle artifacts